syntax = "proto2";

message helloworld   
{
    required int32 id=1;   
    optional string str=2;

    enum PhoneType {               
        MOBILE = 0;
        HOME = 1;
        WORK = 2;
    }

    message PhoneNumber {         
        optional string number = 1;
        optional PhoneType type = 2;
    }
    
    repeated PhoneNumber phones = 4;
}

// The greeting service definition.
service ServiceHello {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}
}
 
// The request message containing the user's name.
message HelloRequest {
  required string name = 1;
}
// The response message containing the greetings
message HelloReply {
  required string message = 1;
}
